### DISCUSSION TITLE
bash tab autocompletion not working

### DISCUSSION THREAD

**Discussion by gothicVI** [git-lfs/git-lfs#5020] [Q&A] [answered]
Hi all,

running Debian 11 here with
```bash
$ git --version 
git version 2.30.2
$ git-lfs --version
git-lfs/3.1.4 (GitHub; linux amd64; go 1.17.8)
```

Tab auto completion works perfectly fine for git itself, however, `git lfs <tab><tab>` or `git-lfs <tab><tab>` will only suggest the files in the current repository, but not any options of git lfs.
Is this intended behavior or is the a bash completion file I'd need to source?

Thanks a lot.

Cheers.

**Comment by chrisd8088** [comment, marked as answer]
I'm not aware of a Bash completion file for Git LFS, myself, but it doesn't mean there isn't one out there.

I believe there was some work years ago in git-lfs/git-lfs#1133 which never reached a successful conclusion, and `spf13/cobra` [provides](https://github.com/git-lfs/git-lfs/blob/315a79216cbaec35b38abdca853ed4cd82389b78/vendor/github.com/spf13/cobra/bash_completions.md) some utilities to help.

If anyone wanted to work on this again, it would be delightful!

**Reply by gothicVI** [reply]
Thanks for your answer.  I wasn't aware that git lfs came without completion from the get go.
